src/main/java/org/javarosa/core/model/TriggerableDag.java

```diff
@@ -345,10 +345,11 @@ public void finalizeTriggerables(FormInstance mainInstance, EvaluationContext ev
             if (deps.contains(qt))
                 throwCyclesInDagException(deps);
 
-            for (QuickTriggerable qu : deps) {
-                QuickTriggerable[] edge = {qt, qu};
-                partialOrdering.add(edge);
-            }
+            if (qt.canCascade())
```

Comment:
That's correct. And we need to always compute the dependencies because we want to detect self-references (cycles) among them. 

I studied the impact of always computing dependencies, and there should be no noticeable difference because the nodes that didn't get their dependencies computed before were nodes that didn't have descendants anyway. It could be good to double check this assumption, though.

Reply:
I agree -- the triggerable types that don't cascade are readonlys and requireds and now the values that their expressions reference will be identified to catch self-references. There's no change when it comes to calculates and relevants.